netmiko/mellanox/mellanox_mlnxos_ssh.py

```diff
@@ -1,10 +1,35 @@
+"""Mellanox MLNX-OS Switch support."""
 from __future__ import unicode_literals
 from netmiko.cisco_base_connection import CiscoSSHConnection
 from netmiko import log
 import time
 
 
-class MellanoxSSH(CiscoSSHConnection):
+class MellanoxMlnxosSSH(CiscoSSHConnection):
+    """Mellanox MLNX-OS Switch support."""
+
+    def check_enable_mode(self, check_string="#"):
+        """Check if in enable mode. Return boolean."""
+        return super(MellanoxMlnxosSSH, self).check_enable_mode(
+            check_string=check_string
+        )
+
+    def enable(self, ena_cmd="enable", pattern="#"):
+        """Enter into enable mode."""
+        output = ""
+        if not self.check_enable_mode():
+            self.write_channel(self.normalize_cmd(ena_cmd))
+            output = self.read_until_pattern(pattern=pattern)
```

Comment:
I see you never send the `secret` so does Mellanox just transition to `enable` mode with no additional password?

Reply:
It does not ask me for password